IdentityCore/src/oauth2/aad_base/MSIDAADOauth2Factory.m

```diff
@@ -105,7 +105,7 @@ - (BOOL)verifyResponse:(MSIDAADTokenResponse *)response
             
             /* This is a special error case for True MAM,
              where a combination of unauthorized client and MSID_PROTECTION_POLICY_REQUIRED should produce a different error */
-            MSIDErrorCode oauthErrorCode = MSIDErrorCodeForOAuthError(response.error, MSIDErrorServerOauth);
+            MSIDErrorCode oauthErrorCode = MSIDErrorCodeForOAuthErrorWithSubErrorCode(response.error, MSIDErrorServerOauth, response.suberror);
```

Comment:
Is the change in this file needed? At the brief look, I see that it's called for silent calls and not interactive ones. I don't think it makes sense to return user cancel error for silent token acquisitions. 

I do see that for interactive calls, [MSIDWebOAuth2AuthCodeResponse.m](https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/pull/1172/files#diff-f61b05eea2aceaf96141e9b10674fde676b2c65450d9be93e2d9db9015129889) already handles the response.